ruby/lib/cucumber/ci_environment.rb

```diff
@@ -23,19 +23,28 @@ def detect(ci_environment, env)
         name: ci_environment['name'],
         url: url,
         buildNumber: evaluate(ci_environment['buildNumber'], env),
-        git: {
-          remote: remove_userinfo_from_url(evaluate(ci_environment['git']['remote'], env)),
-          revision: evaluate(ci_environment['git']['revision'], env),
-          branch: evaluate(ci_environment['git']['branch'], env),
-        }
       }
-      tag = evaluate(ci_environment['git']['tag'], env)
-      if tag
-        result[:git][:tag] = tag
-      end
+
+      detected_git = detect_git(ci_environment, env)
+      result[:git] = detected_git if detected_git
       result
     end
 
+    def detect_git(ci_environment, env)
+      revision = evaluate(ci_environment['git']['revision'], env)
+      return nil if revision.nil?
+
+      git_info = {
+        remote: remove_userinfo_from_url(evaluate(ci_environment['git']['remote'], env)),
+        revision: evaluate(ci_environment['git']['revision'], env),
+        branch: evaluate(ci_environment['git']['branch'], env),
```

Comment:
I think we should conditionally assign `:branch` in the same way we conditionally assign `:tag`

Reply:
We might need 2 new tests:

- One where remote is set and revision is not
- One where revision is set and remote is not

Reply:
> We might need 2 new tests:
> 
>     * One where remote is set and revision is not
> 
>     * One where revision is set and remote is not

What do we do in those cases? We do not set git even if one of the two data exists?

Reply:
Correct - see details in https://github.com/cucumber/ci-environment/issues/36